src/storage/mutate/AddEdgesAtomicProcessor.cpp

```diff
@@ -4,140 +4,92 @@
-using ChainId = std::pair<PartitionID, PartitionID>;
-
 void AddEdgesAtomicProcessor::process(const cpp2::AddEdgesRequest& req) {
     propNames_ = req.get_prop_names();
     spaceId_ = req.get_space_id();
-
-    auto stVidLen = env_->schemaMan_->getSpaceVidLen(spaceId_);
-    if (!stVidLen.ok()) {
-        LOG(ERROR) << stVidLen.status();
+    auto code = getSpaceVidLen(spaceId_);
+    if (code != cpp2::ErrorCode::SUCCEEDED) {
         for (auto& part : req.get_parts()) {
             pushResultCode(cpp2::ErrorCode::E_INVALID_SPACEVIDLEN, part.first);
         }
         onFinished();
         return;
     }
-    vIdLen_ = stVidLen.value();
-    processByChain(req);
-}
 
-void AddEdgesAtomicProcessor::processByChain(const cpp2::AddEdgesRequest& req) {
-    std::unordered_map<ChainId, std::vector<KV>> edgesByChain;
-    std::unordered_map<PartitionID, cpp2::ErrorCode> failedPart;
-    // split req into chains
-    for (auto& part : req.parts) {
-        auto localPart = part.first;
-        for (auto& edge : part.second) {
-            auto stPartId = env_->metaClient_->partId(spaceId_, edge.key.dst.getStr());
-            if (!stPartId.ok()) {
-                failedPart[localPart] = cpp2::ErrorCode::E_SPACE_NOT_FOUND;
-                break;
-            }
-            auto remotePart = stPartId.value();
-            ChainId cid{localPart, remotePart};
-            if (FLAGS_trace_toss) {
-                auto& ekey = edge.key;
-                LOG(INFO) << "ekey.src.hex=" << folly::hexlify(ekey.src.toString())
-                          << ", ekey.dst.hex=" << folly::hexlify(ekey.dst.toString());
-            }
-            auto key = TransactionUtils::edgeKey(vIdLen_, localPart, edge.get_key());
-            std::string val;
-            auto code = encodeSingleEdgeProps(edge, val);
-            if (code != cpp2::ErrorCode::SUCCEEDED) {
-                failedPart[localPart] = code;
-                break;;
-            }
-            edgesByChain[cid].emplace_back(std::make_pair(std::move(key), std::move(val)));
-        }
-    }
-
-    if (!failedPart.empty()) {
-        for (auto& part : failedPart) {
-            pushResultCode(part.second, part.first);
+    auto vidTypeStatus = env_->metaClient_->getSpaceVidType(spaceId_);
+    if (!vidTypeStatus) {
+        for (auto& part : req.get_parts()) {
+            pushResultCode(cpp2::ErrorCode::E_INVALID_VID, part.first);
         }
         onFinished();
         return;
     }
+    spaceVidType_ = std::move(vidTypeStatus).value();
+    CHECK_LT(req.get_parts().begin()->second.front().get_key().get_edge_type(), 0);
 
-    CHECK_NOTNULL(env_->indexMan_);
-    auto stIndex = env_->indexMan_->getEdgeIndexes(spaceId_);
-    if (stIndex.ok()) {
-        if (!stIndex.value().empty()) {
-            processor_.reset(AddEdgesProcessor::instance(env_));
-            processor_->indexes_ = stIndex.value();
-        }
+    processInEdges(const_cast<cpp2::AddEdgesRequest&>(req));
```

Comment:
why `processInEdges `not use `const cpp2::AddEdgesRequest& req` ?

Reply:
Doodle suggest some member can be moved into chain processor, that's why I need to remove that 'const'